parent/pom.xml

```diff
@@ -1278,5 +1309,25 @@
             </build>
         </profile>
 
+        <profile>
+            <id>checkstyle</id>
+
+            <activation>
+                <activeByDefault>true</activeByDefault>
```

Comment:
Is it really have to be activated by default? That can significantly increase  build times. Let's activate it only in corresponding cases like intentional checkstyle run.

Reply:
I've measured the increasing build time - it's about +10% with the enabled checkstyle plugin. 
Fixed too.